launchlib/launcher.go

```diff
@@ -276,12 +281,21 @@ func delim(str string) string {
 	return fmt.Sprintf("%s%s%s", TemplateDelimsOpen, str, TemplateDelimsClose)
 }
 
-func createJvmOpts(combinedJvmOpts []string, customConfig *CustomLauncherConfig, logger io.WriteCloser) []string {
+func createJvmOpts(combinedJvmOpts []string, customConfig *CustomLauncherConfig, logger io.WriteCloser) ([]string, error) {
 	if isEnvVarSet("CONTAINER") && !customConfig.DisableContainerSupport && !hasMaxRAMOverride(combinedJvmOpts) {
 		_, _ = fmt.Fprintln(logger, "Container support enabled")
-		combinedJvmOpts = filterHeapSizeArgs(combinedJvmOpts)
-		combinedJvmOpts = ensureActiveProcessorCount(combinedJvmOpts, logger)
-		return combinedJvmOpts
+		if customConfig.Experimental.ContainerV2 {
+			jvmOptsWithUpdatedHeapSizeArgs, err := filterHeapSizeArgsV2(combinedJvmOpts)
+			if err != nil {
+				combinedJvmOpts = filterHeapSizeArgs(combinedJvmOpts)
+			} else {
+				combinedJvmOpts = jvmOptsWithUpdatedHeapSizeArgs
+			}
+		} else {
+			combinedJvmOpts = filterHeapSizeArgs(combinedJvmOpts)
+		}
+		combinedJvmOpts = ensureActiveProcessorCount(customConfig, combinedJvmOpts, logger)
```

Comment:
I believe this statement is only meaningful in the `else` block in on 295 when `customConfig.Experimental.ContainerV2` is false. Should we move it there?